ofdb-entities/src/nonce.rs

```diff
@@ -116,13 +116,80 @@ impl EmailNonce {
     }
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ReviewNonce {
+    pub place_id: Id,
+    pub place_revision: Revision,
+    pub nonce: Nonce,
+}
+
+#[derive(Debug)]
+pub enum ReviewNonceDecodingError {
+    Bs58(bs58::decode::Error),
+    Utf8(std::string::FromUtf8Error),
+    TooShort(ActualTokenLen),
+    Parse(NonceString, NonceParseError),
+    Revision,
+}
+
+const MIN_REVISION_STR_LEN: usize = 1;
+
+impl ReviewNonce {
+    pub fn encode_to_string(&self) -> String {
+        let nonce = self.nonce.to_string();
+        debug_assert_eq!(Nonce::STR_LEN, nonce.len());
+        let place_id = self.place_id.as_str();
+        let revision_value = RevisionValue::from(self.place_revision);
+        let revision_string = revision_value.to_string();
+        let mut concat =
```

Comment:
Why no readable `format!()` expression?

Reply:
...because I copy & paste the code from the `EmailNonce` implementation and I thought someone has thought of something clever :stuck_out_tongue_winking_eye: